code/modules/client/mouse_pointer/mouse_pointer_definitions.dm

```diff
@@ -0,0 +1,31 @@
+/decl/mouse_pointer
+	abstract_type = /decl/mouse_pointer
+	/// Icon to set on the client for this cursor.
+	var/list/icons
+
+/decl/mouse_pointer/Initialize()
+	. = ..()
+	if(icons && !islist(icons))
+		icons = list(icons)
+
+/decl/mouse_pointer/validate()
+	. = ..()
+	if(length(icons))
+		var/static/list/check_states = list(
+			"",
+			"over",
+			"drag",
+			"drop",
+			"all"
+		)
+		for(var/icon in icons)
+			for(var/check_state in check_states)
+				if(!check_state_in_icon(check_state, icon))
+					. += "missing state '[check_state]' from icon '[icon]'"
```

Comment:
i wonder if it would be possible to make a helper that checks multiple icon states faster? either through list intersection or some other method. not necessary for functionality but i figure this is probably done elsewhere